Use wp-sitemap(.*).xml instead of sitemap(.*).xml

pfefferle opened the pull request:
### Issue Number
#128 

### Description
Use `wp-sitemap(.*).xml` instead of `sitemap(.*).xml` and add a possibility to filter the filename prefix.

### Type of change
Please select the relevant options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Enhancement (change which improves an existing feature. E.g., performance improvement, docs update, etc.)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Acceptance criteria
- [X] My code follows WordPress coding standards.
- [X] I have performed a self-review of my own code.
- [ ] If the changes are visual, I have cross browser / device tested.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] My changes generate no new warnings.
- [ ] I have added test instructions that prove my fix is effective or that my feature works.

pfefferle committed:
implement `core_sitemaps_filename_prefix()`

fix #128

pfefferle committed:
fix placeholder

pfefferle committed:
fix broken method call

swissspidy commented on inc/functions.php:
```diff
@@ -49,6 +49,20 @@ function core_sitemaps_get_sitemaps() {
+ *
+ * @return string Prefix.
+ */
+function core_sitemaps_filename_prefix() {
```
`filename` might be misleading because they're not actually files. Maybe just `core_sitemaps_prefix`? Or perhaps `core_sitemaps_sitemap_prefix`

pfefferle replied:
`core_sitemaps` will be removed when merging into core!?! If so, I would prefer `core_sitemaps_sitemap_prefix`.

swissspidy replied:
That hasn't been decided yet. We might also just replace `core_sitemaps` with `wp_sitemaps` or end up using something like `wp_sitemap_prefix` in the end. But this doesn't matter for the sake of adding it to the plugin 🙂

pfefferle committed:
replace `filename` with `sitemap`

https://github.com/GoogleChromeLabs/wp-sitemaps/pull/129#pullrequestreview-362624791

pfefferle committed:
use core_sitemaps_sitemap_prefix

pfefferle committed:
fix bracket issue

pfefferle committed:
Merge branch 'master' into patch-128

kraftbj commented:
I'd suggest also hooking into pre_handle_404 while a feature plugin and patch `handle_404` during Core merge. If someone attempts /sitemap.xml (or other supported formats) and it would 404, let's redirect them to the wp-sitemap.xml.

That would help discoverability of the sitemap ("sitemap.xml" being pretty common, like "feed" is for RSS) while not conflicting with existing solutions.

I'm happy to work on a patch for this to be applied to this branch (or master if post merge).

swissspidy commented:
Added an alternative PR without the filter here: #135

pfefferle commented:
closed in favor of #135
